pexec/managed_process.go

```diff
@@ -109,7 +109,23 @@ func (p *managedProcess) ID() string {
 func (p *managedProcess) Status() error {
```

Comment:
this api seems unwieldy - if anyone wants to extract the exit code, they have to parse the error. I feel like an enum might make more sense (states for running, haven't started, exited, signaled, etc) if we want to keep `Status()` as the  method name. We would lose exit code, but not the most important for our purposes. Or just `isAlive() bool` to simplify it even more

Reply:
great points, I agree. Taking this all the way to clear enums and the parsing/handling that would be necessary to map things correctly across architectures, OSs, etc.

I changed it to `IsRunning()`. This also covers the case where the `cmd` is not set for some reason.